### ISSUE TITLE
[HOLD for payment 2023-05-25] [$2000] Android - App adds additional letters after selecting a suggested emoji and pressing Enter (gboard)

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#16106]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open the Android app
2. Open any chat
3. Type colon and 2 letters to get emoji suggestions (eg: ':he',':jo',':bl')
4. Select any one suggested emoji and press enter

## Expected Result:
You should go to the next line (this is typical "Enter" behaviour)

## Actual Result:
App goes to next line but adds previously typed 2 letters after the emoji (see video below)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.88-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** 

https://user-images.githubusercontent.com/43996225/226214883-7c2c6493-01c5-4e54-a2c3-36c6c329610a.mp4

https://user-images.githubusercontent.com/43996225/226214886-2e91473f-7288-485a-a7af-58629075b170.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dhanashree-sawant
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1679074253884739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010623c6559be6640a</li>
        <li>Upwork Job ID: 1638426418352893952</li>
        <li>Last Price Increase: 2023-05-08</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @davidcardoza (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by davidcardoza** [comment]
I am heading out of office for the next 3 days, reassigning.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @jliexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by jliexpensify** [comment]
What a weird bug, I can repro on v88 (Pixel 3a).

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~010623c6559be6640a

**Comment by MelvinBot** [comment]
Current assignee @jliexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mollfpr (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MariaHCD (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by kaushiktd** [comment]
**Please re-state the problem that we are trying to solve in this issue.**
Android - App adds additional letters after selecting a suggested emoji and pressing Enter(gboard)

**What is the root cause of that problem ?**
As per `updateComment` in the below file
https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionCompose.js#L518
if you are updating emojis using **: ear :** (start and end with colon without space) it will show proper emojis without any additional letters but if you are updating **:ear** (only start with colon and select from suggestion list), it will update 
as an emoji icon + ear like (👂 ear )
It will show the additional letter if you are updating emojis without closing colon like **(:ear)**

**What changes do you think we should make in order to solve the problem?**
You need to update `insertSelectedEmoji` function in the below file
https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionCompose.js#L426
you need to replace line#432 from

`this.updateComment(`${commentBeforeColon}${emojiCode} ${commentAfterColonWithEmojiNameRemoved}`, true);`

to below mention line

`this.updateComment(`:${emojiObject.name}:`, true)`

(pass emojis name enclose with colon instead of `emojiCode`)

**Comment by mollfpr** [comment]
I can reproduce it with the Pixel 2 simulator but not the SM Flip 3 physical device.

---

@kaushiktd I try your solution, but it's still not working, and your proposal does not explain why selecting the emoji with `:jo` or `:ea` is adding the text again but not with `:joy` or `:ear`.

Here I'm applying your solution.

https://user-images.githubusercontent.com/25520267/227272387-5cdd6a72-0934-415e-b0ca-983f391dada2.mov

**Comment by kaushiktd** [comment]
@mollfpr See the video here,
https://drive.google.com/file/d/1e1zo_yAowYTd9pIuR4YmEUGm3MNn9BkQ/view?usp=share_link

**Comment by kaushiktd** [comment]
@mollfpr  You are correct. Will come up with a new solution. Thanks.

**Comment by kaushiktd** [comment]
@mollfpr updated my proposal,
https://github.com/Expensify/App/issues/16106#issuecomment-1480687210

**Comment by mollfpr** [comment]
@kaushiktd Thanks for the update! I see that you are removing the extra selection for the space, although it's fixing this issue we are missing the extra space after the emoji which is important.

**Comment by kaushiktd** [comment]
Hope you had a good weekend @mollfpr 
I have updated the proposal after your input.

**Comment by MariaHCD** [comment]
Not overdue, proposals are currently being evaluated by @mollfpr

**Comment by mollfpr** [comment]
I'm not making it for today; I'll review this in the morning. Thank you!

**Comment by MariaHCD** [comment]
@mollfpr Were you able to review the updates to the proposals here? :)

**Comment by mollfpr** [comment]
@MariaHCD Sorry, I forgot about this, and thank you for the ping! I'll review this tonight.

**Comment by mollfpr** [comment]
@allroundexperts If this issue is expected on the RN project, I prefer we come up with solution to fix it on the upstream.

**Comment by mollfpr** [comment]
@allroundexperts Do you suggesting to use `react-native-text-input-reset` or else? As you said, the `react-native-text-input-reset` is not maintained well. It's not worth installing a new library to solve an issue not happening in most Android devices.

The recent proposal from @kaushiktd is fixing the issue, but I need more context about the solution.

**Comment by kaushiktd** [comment]
> @kaushiktd Could you explain why we need to call `this.setTextInputShouldClear(true);` to clear the composer value? and why it's not happen in all Android device?

Sure. When the user types a message and send it, `submitForm()` is called. This function calls another function  `prepareCommentAndResetComposer()`, which is clearing the composer value, but when the user starts typing with colon (:) and chooses emojis with `insertSelectedEmoji()` function, it converts composer value to emojis code but it is not clearing text value, so it will show extra text value after emojis. We need to call t`his.setTextInputShouldClear(true);` to clear composer.

**Comment by kaushiktd** [comment]
> I did not realise that I was posting a duplicate proposal. Can you please share a link where you proposed the same thing? Apologies if my proposal caused inconvenience. I'll be happy to revoke it in your favour after checking the link

It seems you misunderstood my comment, brother! 🙂 I said, I first came up with the same solution as you did but upon working more on it I found that it is not a good solution. I did not say you duplicated it. I never proposed that solution here after making myself certain about it but upon checking recently I found that you proposed the same solution that I refrained myself to propose. So I said I CONSIDERED that solution, not proposed it here. 👍🏻

**Comment by kaushiktd** [comment]
> Ah. No worries. Thanks fir the clarification!

No problem!

**Comment by mollfpr** [comment]
I appreciate that your both respect each other and working on this issue to find a better solution @kaushiktd @allroundexperts 👍

**Comment by mollfpr** [comment]
@kaushiktd I test your proposal on Pixel 2 simulator, and it seems to be working to solve the issue, but it's clear the input on iOS.

https://user-images.githubusercontent.com/25520267/228849092-907264f5-0c51-42ba-b826-792c3e81ae80.mov

**Comment by kaushiktd** [comment]
@mollfpr On it.

**Comment by jliexpensify** [comment]
Hmm, this was auto-doubled, even though we removed that 2 weeks ago...odd. Changing back to $1000 @MariaHCD and @mollfpr

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $1000

**Comment by mollfpr** [comment]
Thanks for the update @kaushiktd!

We do not allow using `Platform.OS == CONST.PLATFORM.ANDROID`. If we need something platform-specific, we should split it into platform-specific files.

There's a missing context of why this issue only occurred in several Android devices, and why it's only occurring when selecting the emoji by typing `:sm` not `:smi`.

**Comment by kaushiktd** [comment]
> We do not allow using `Platform.OS == CONST.PLATFORM.ANDROID`. If we need something platform-specific, we should split it into platform-specific files.

I did not know that. Since it was working in iOS as usual and only showed problems in android, I suggested that solution. Let me see what else can be done there.

**Comment by MelvinBot** [comment]
@MariaHCD @mollfpr @jliexpensify this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by kaushiktd** [comment]
@mollfpr  proposal updated

**Comment by MariaHCD** [comment]
@mollfpr looks like we've got an updated proposal from @kaushiktd! Do jump in on that when you can, thanks!

**Comment by mollfpr** [comment]
Thank you for the update @kaushiktd 

Can you explain what the custom package does and why we need to create one? Is `this.textInput.clear()` not doing the job?

> There's a missing context of why this issue only occurred in several Android devices, and why it's only occurring when selecting the emoji by typing :sm not :smi.

We still missing the root cause is.

**Comment by mollfpr** [comment]
@allroundexperts Same as @kaushiktd, I still miss the root cause because it is not happening on all Android devices and if something is broken in the upstream RN, we should fix it upstream.

**Comment by kaushiktd** [comment]
> Thank you for the update @kaushiktd
> 
> Can you explain what the custom package does and why we need to create one? Is `this.textInput.clear()` not doing the job?
> 
> > There's a missing context of why this issue only occurred in several Android devices, and why it's only occurring when selecting the emoji by typing :sm not :smi.
> 
> We still missing the root cause is.

@mollfpr  Some android keyboards like gboard, TouchPal etc are not clearing the textInput value properly and repeats the last word from the previously typed string. 
This custom package is resetting or clearing the value of a text input component in a React Native application properly. 

this.textInput.clear() clear sthe value for textInput box in all platforms like ios, android and web.

**Comment by MelvinBot** [comment]
@MariaHCD @mollfpr @jliexpensify this issue is now 3 weeks old. There is one more week left before this issue <a href='https://stackoverflowteams.com/c/expensify/questions/15233/15234#15234'>breaks WAQ</a> and will need to go internal. What needs to happen to get a PR in review this week? Please create a thread in #expensify-open-source to discuss. Thanks!

**Comment by MelvinBot** [comment]
@MariaHCD, @mollfpr, @jliexpensify Eep! 4 days overdue now. Issues have feelings too...

**Comment by MelvinBot** [comment]
@MariaHCD, @mollfpr, @jliexpensify Eep! 4 days overdue now. Issues have feelings too...

**Comment by MariaHCD** [comment]
Not overdue, we're still evaluating proposals and determining the root cause of the issue here

**Comment by mollfpr** [comment]
@kaushiktd Sorry for the wait! Could you open a discussion about your latest proposal in `#expensify-open-source`? I'm still undecided about adding a custom native package as our best solution.

cc @MariaHCD

**Comment by kaushiktd** [comment]
@mollfpr I just started one here. 
https://github.com/kaushiktd/Expensify.cash/discussions/1

Hope I did it right as I did that before. 😅

**Comment by mollfpr** [comment]
@kaushiktd I mean, create a thread in our Slack channel 😅

Have you joined the Slack channel?

**Comment by mollfpr** [comment]
It can be, so I think it's a Slack discussion. Because I haven't seen any implementation with a custom native package while adding a new library is something we need to discuss with a broader audience to get the approval.

There's also an interesting statement from the `react-native-text-input-reset` maintainer.

>This package provides a temporary hack around a bug found in some android keyboards so that the contents of a TextInput can be properly cleared. https://issuetracker.google.com/issues/36928977
>
>It's likely this will be solved in a future release of React Native. https://github.com/facebook/react-native/pull/12462

https://github.com/nikolaiwarner/react-native-text-input-reset/blob/51fc3605a9f3847d133c434f4de1c49537098e97/README.md?plain=1#L4-L9

---

Actually, it still bothers me why type `:sm` I can reproduce the issue but not with `:smi`.

**Comment by mollfpr** [comment]
@allroundexperts Yes. Anything more than two letter is producing the issue.

**Comment by kaushiktd** [comment]
@mollfpr please send me link to the slack.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @abekkala (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by jliexpensify** [comment]
@abekkala - apologies, had to reassign as I am OOO for the next 2 weeks. At this point, we're still looking into proposals so no action from you is needed!

**Comment by mollfpr** [comment]
Move the discussion to `expensive-open-source` https://expensify.slack.com/archives/C01GTK53T8Q/p1681310437628459

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by mollfpr** [comment]
We have an open discussion on the channel and waiting for proposals.

**Comment by MelvinBot** [comment]
@abekkala @MariaHCD @mollfpr this issue is now 4 weeks old and preventing us from maintaining WAQ, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is today
- If no proposals meet that standard, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #expensify-open-source

Thanks!

**Comment by MelvinBot** [comment]
Current assignee @mollfpr is eligible for the Internal assigner, not assigning anyone new.

**Comment by MariaHCD** [comment]
This does not need to be internal, discussion in Slack is still ongoing: https://expensify.slack.com/archives/C01GTK53T8Q/p1681310437628459

**Comment by MelvinBot** [comment]
Current assignee @abekkala is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @MariaHCD is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Upwork job price has been updated to $2000

**Comment by mallenexpensify** [comment]
Not sure why the price hasn't auto-increased the past month, bumped to $2K to help prioritize the fix (and it doesn't appear to be a quick fix).

**Comment by MariaHCD** [comment]
Re-assigning the CME here since I'm going OOO for two weeks.

**Comment by MelvinBot** [comment]
Current assignee @abekkala is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Current assignee @mollfpr is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @johnmlee101 (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by mollfpr** [comment]
Just putting my thoughts from what we have till now. The current solution to reset the text input in Android resolves the issue. We don't have a clear explanation about the root cause, probably due to setting the input value dynamically creating the issue, and we already have the workaround with `this.setTextInputShouldClear`.

Now we are looking for an excellent approach to reset the input but only for Android. The current proposal is:
1. Create a custom native package, and use it from `NativeModule` to expose the reset input function in the Android
2. Fork `react-native-text-input-reset`

cc @johnmlee101

**Comment by mollfpr** [comment]
>In node_modules/react-native-text-input-reset/android/build.gradle, update compile on line 36 to implementation.
Remove the following block in file node_modules/react-native-text-input-reset/android/src/main/java/com/nikolaiwarner/RNTextInputReset/RNTextInputResetModule.java:


@allroundexperts I'm curious about this part; why should we modify the package again? Can we just use the original function from the library?

**Comment by MelvinBot** [comment]
@johnmlee101, @abekkala, @mollfpr Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by mollfpr** [comment]
@johnmlee101 What about adding a custom native package? I'm unsure because we don't have one implemented in our App.

**Comment by johnmlee101** [comment]
You're talking about
> Create a custom native package, and use it from NativeModule to expose the reset input function in the Android

Right? Could you detail what that would look like if implemented?

**Comment by johnmlee101** [comment]
Ah yeah, okay. If that's the only function we require and its no longer maintained, then I think its fine to move that function only to our code.

**Comment by kaushiktd** [comment]
@johnmlee101 if you look at my proposal, you'll realise I've already given similar solution with the same path where you need to add just two files, which I've also given and it will be done and it will no longer required to install npm.

**Comment by kaushiktd** [comment]
I've already clarified everything in one of my comments a while ago, my friend. Thanks.

**Comment by mollfpr** [comment]
@allroundexperts Could you update your proposal based on your [comment](https://github.com/Expensify/App/issues/16106#issuecomment-1525968637)?

**Comment by mollfpr** [comment]
I have concerns regarding using the native module. Based on the RN doc about Android native modules https://reactnative.dev/docs/native-modules-android. Soon android native module will be deprecated if the new architecture is stable.

>Native Module and Native Components are our stable technologies used by the legacy architecture. They will be deprecated in the future when the New Architecture will be stable. The New Architecture uses [Turbo Native Module](https://reactnative.dev/docs/the-new-architecture/pillars-turbomodules) and [Fabric Native Components](https://reactnative.dev/docs/the-new-architecture/pillars-fabric-components) to achieve similar results.

We want the solution to work in the new architecture.

cc @johnmlee101

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by mollfpr** [comment]
Still reviewing the proposals and am also open to other proposals.

**Comment by johnmlee101** [comment]
Do we have plans to migrate to the new architecture today?

**Comment by mollfpr** [comment]
No, as far as I know. IIRC, we have an E/App cloned where we try to migrate to the fabric.

**Comment by johnmlee101** [comment]
Yeah my only concern is the future-proof architecture docs are all marked as Experimental, so I think its fine to work with what is stable now, even if theres a chance it will be deprecated in the future

**Comment by mollfpr** [comment]
@johnmlee101 For the native module implementation, you can check the detail in @kaushiktd [proposal](https://github.com/Expensify/App/issues/16106#issuecomment-1480687210).

I already test the solution, and it fixed the issue.

**Comment by michaelhaxhiu** [comment]
@mollfpr does that mean we are selecting @kaushiktd s proposal?

**Comment by mollfpr** [comment]
@michaelhaxhiu @michaelhaxhiu Sounds good to me!

**Comment by kaushiktd** [comment]
**We can add the same function (the one which react-native-text-input-reset is exposing), 
into our app's Android src folder as well here. This will result in NPM package no longer being needed.**

If you are using the npm package, you need to install it first and then call the below function,

```
TextInputReset.resetKeyboardInput(
findNodeHandle(this.textInput));
```

to reset the input value from the keyboard in the below file.
https://github.com/allroundexperts/Expensify/blob/10d3d52e993303db477478d323af925f4d439acf/src/pages/home/report/ReportActionCompose.js#L444

Above comment is suggesting to create a custom package on the below location,
https://github.com/Expensify/App/tree/main/android/app/src/main/java/com/expensify/chat

### In my proposal
I have already created RNTextInputResetPackage.java and RNTextInputResetModule.java files on the same mentioned location, and you do not need to call any function on the RN code side. So that is a partial copy of what I've already suggested but in another words.

I don't think any of the assignee contributors would ever do such favours to anyone here.🙂

**Comment by kaushiktd** [comment]
Just applied on upwork! @johnmlee101 @mollfpr

**Comment by abekkala** [comment]
not overdue. John has chosen the proposal. Waiting on PR

**Comment by kaushiktd** [comment]
@abekkala @johnmlee101  I'm waiting for one of you guys to accept my proposal on upwork. Or should I first do the PR?

**Comment by mollfpr** [comment]
@kaushiktd will be better to start work on the PR so we will have a chance to get the timeline bonus.

**Comment by kaushiktd** [comment]
Okay, PR will be ready in the next 24 hrs OR earlier! This is my first time so I hope you guys can guide me if needed. Though I have checked everything in guidelines on how this works step by step. Thank you!

**Comment by kaushiktd** [comment]
PR is ready for review @johnmlee101 @mollfpr

**Comment by abekkala** [comment]
Looks like PR was deployed to staging yesterday. 

@johnmlee101 should that have triggered the auto-update to the title for  payment date?

**Comment by kaushiktd** [comment]
Any update on this, @mollfpr @johnmlee101 ?

**Comment by mollfpr** [comment]
> If no regressions arise, payment will be issued on 2023-05-25. 🎊

@kaushiktd Hang tight!

**Comment by abekkala** [comment]
@mollfpr can you also complete the checklist above? that will need to be done prior to payment being made

**Comment by mollfpr** [comment]
> [@mollfpr] The PR that introduced the bug has been identified. Link to the PR:
 [@mollfpr] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:

It is a bug, but it's not reproduced in all Android devices, and no offending PR is causing this issue.

> [@mollfpr] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

Since it's not reproduced in all Android devices, we can't catch this kind of bug in our development process.

> [@mollfpr] Determine if we should create a regression test for this bug.

I'm inclined to have this regression test.

>[@mollfpr] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.

Propose regression test

1. Open the App
2. Navigate to any existing chat
3. type `:sm` and select emojis from the suggested list
4. Press the return / enter button from the keyboard.
5. **Verify** that the composer input added a new line and that no text was added after the emoji
6. 👍 or 👎

**Comment by abekkala** [comment]
### Payments to be made on May 25:

- Issue reported by: @dhanashree-sawant [$250]
- Selected Proposal for fix: @kaushiktd [$2000] + [$500 Bonus, if no regressions]
- C+ Review: @mollfpr [$2000] + [$500 Bonus, if no regressions]

**Comment by kaushiktd** [comment]
@abekkala  the job has not been awarded to me yet! Just for your info. Not sure if you usually award the job at the time of making a payment

**Comment by kaushiktd** [comment]
Any update on this so far? @mollfpr @abekkala

**Comment by mollfpr** [comment]
@abekkala will proceed with the payment when they are online.

**Comment by kaushiktd** [comment]
But I have not been assigned the job on Upwork yet!

**Comment by mollfpr** [comment]
@kaushiktd Me neither! Only CME/BZ can create the job and proceed the payment, so wait patiently until they are online.

**Comment by abekkala** [comment]
All offers were sent - once accepted I'll pay them! 🎉

**Comment by abekkala** [comment]
@dhanashree-sawant  paid and contract ended. Thank you! 🎉

**Comment by abekkala** [comment]
@kaushiktd paid and contract ended. Thank you! 🎉

**Comment by mollfpr** [comment]
@abekkala accepted 🎉
